server/test/api/costCenter.js

```diff
@@ -0,0 +1,105 @@
+/*global describe, it, beforeEach, process*/
+
+var chai = require('chai');
+var chaiHttp = require('chai-http');
+var expect = chai.expect;
+chai.use(chaiHttp);
+
+var helpers = require('./helpers');
+helpers.configure(chai);
+
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = 0;
+
+var url = 'https://localhost:8080';
+var user = { username : 'superuser', password : 'superuser', project: 1};
+
+describe('The /cost_center API endpoint', function () {
+ var agent = chai.request.agent(helpers.baseUrl);
+ var newCostCenter = {
+    project_id : 1,
+    //id : 200,
+    text : 'tested cost',
+    note : 'test inserted',
+    is_principal : 1
+  };
+
+  var DELETABLE_COST_CENTER_ID = 2;
+
+  var FETCHABLE_COST_CENTER_ID = 1;
+
+   // throw errors
+  beforeEach(helpers.login(agent));
+
+    it(' A GET /cost_centers?list=full returns a list of cost centers', function () {
+      return agent.get('/cost_centers?full=1')
+        .then(function (res) {
+          expect(res).to.have.status(200);
+          expect(res.body).to.not.be.empty;
+          expect(res.body).to.have.length(2);
+        })
+       .catch(helpers.handler);
+    });
+
+  it(' A GET /cost_centers returns a list of cost centers', function () {
+    return agent.get('/cost_centers')
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res.body).to.not.be.empty;
+        expect(res.body).to.have.length(2);
+      })
+      .catch(helpers.handler);
+ });
+  it(' A GET /cost_center/:id returns one cost center', function () {
+    return agent.get('/cost_centers/'+ FETCHABLE_COST_CENTER_ID)
+      .then(function (res) {
+        expect(res).to.have.status(200);
+        expect(res.body).to.not.be.empty;
+        expect(res.body.id).to.be.equal(FETCHABLE_COST_CENTER_ID);
+      })
+      .catch(helpers.handler);
+ });
+  it('A POST /cost_centers will add a cost center', function () {
+    return agent.post('/cost_centers')
+      .send(newCostCenter)
+      .then(function (res) {
+        expect(res).to.have.status(201);
+        expect(res).to.be.json;
+        expect(res.body).to.not.be.empty;
+        expect(res.body.id).to.be.defined;
+        newCostCenter.id = res.body.id;
```

Comment:
You should check that the record was properly inserted into the database by looking it up via an additional GET request.